lib/pdk/control_repo.rb

```diff
@@ -0,0 +1,45 @@
+require 'pdk'
+
+module PDK
+  module ControlRepo
+    CONTROL_REPO_FILES = %w[Puppetfile environment.conf].freeze
+
+    # Returns path to the root of the Control Repo being worked on.
+    #
+    # An environment.conf is required for a Control Repo, whereas Puppetfile is
+    # optional. Note that a Bolt Project can also be a Control Repo.
+    #
+    # @see https://puppet.com/docs/pe/latest/control_repo.html
+    #
+    # @param strict_check [Boolean] When strict_check is true, only return the path
+    #   if the Control Repo is strictly _only_ a control repository. For example,
+    #   not also a Puppet Bolt project directory  Default is false.
+    #
+    # @return [String, nil] Fully qualified base path to Control Repo, or nil if
+    #   the current working dir does not appear to be within a Control Repo.
+    def control_repo_root(strict_check = false)
+      environment_conf_path = PDK::Util.find_upwards('environment.conf')
+      path = if environment_conf_path
+               File.dirname(environment_conf_path)
+             elsif in_control_repo_root?(Dir.pwd)
+               Dir.pwd
```

Comment:
If we are requiring an `environment.conf` to be present to consider something a control-repo, what additional benefit does this branch get us versus the `find_upwards` search? (`find_upwards` checks in `pwd` first before moving up the tree)

Reply:
`in_control_repo_root?` looks for things that like _like_ a control repo (e.g. Puppetfile) whereas this method looks for something explicit.  TBH this is mostly a copy-paste from the module detection. 

I'm still in two minds about the whole explicit vs fuzzy search.  The docs clearly state that "environment.conf" == Control Repo and Puppet itself will error if it's not present.  Whereas in module land, Puppet will use a directory without a metadata.json happily, but we prefer a metadata.json.

Reply:
@rodjek Thoughts?

Reply:
Had a chat with @rodjek.  And while non-directory based environment control repos do exist, they are not the preferred way.  Also detecting them is problematic as they don't have an easily identifiable thing to search for.

So for this PR  `find_control_repo_root` will look for environment.conf whereas `control_repo_root?` will look for something that _could_ be a control repo. Use cases:

* `find_control_repo_root` would be used in a `pdk validate` operation
* `control_repo_root?` would be used in a `pdk convert` operation